src/uu/csplit/src/csplit_error.rs

```diff
@@ -1,26 +1,28 @@
 use std::io;
 use thiserror::Error;
 
+use uucore::display::Quotable;
+
 /// Errors thrown by the csplit command
 #[derive(Debug, Error)]
 pub enum CsplitError {
     #[error("IO error: {}", _0)]
     IoError(io::Error),
-    #[error("'{}': line number out of range", _0)]
+    #[error("{}: line number out of range", ._0.quote())]
```

Comment:
I'm not sure these should be quoted. GNU apparently doesn't escape them, and these are patterns (e.g. a regex, not file names). On the other hand, it might be nice to have.

Reply:
I'm not familiar with `csplit` but I think I see some escaping going on:
```console
$ csplit /dev/null $'\n'
csplit: ‘\n’: invalid pattern
$ csplit /dev/null $'\xFF'
csplit: ‘\377’: invalid pattern
```

Reply:
I'm not familiar with it either. The example I tried contained a `'`:
`LC_ALL=C csplit tests/fixtures/csplit/numbers50.txt  /9\'$/ {50}` (this is similar to a test in `test_csplit.rs`). It prints `csplit: '/9\'$/': match not found`, which doesn't seem to be the correct. Anyways, I think your change is fine.